### DISCUSSION TITLE
Performance calculation using FLOPS for LAPACK/LAPACKE APIs.

### DISCUSSION THREAD

**Discussion by mopathan** [Reference-LAPACK/lapack#704] [General]
Hi All,
Hope you are doing well and safe.
 
Currently, we are facing difficulty to get the performance calculation (FLOPS count) of each LAPACK/LAPACKE API using FLOP count for verification.
Either we will search in internet for each API and for few APIs, we will get in Intel MKL website. Some we may found and others may not.
 
Can you please help to add performance info as comment in each LAPACK/LAPACKE API source code?
Also request to add performance info in each API documentation in netlib documentation.
http://www.netlib.org/lapack/explore-html/globals_func_s.html

Example, for SSTEQR() API, we have calculation details in MKL link: https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/lapack-routines/lapack-least-squares-and-eigenvalue-problem/lapack-least-square-eigenvalue-problem-computation/symmetric-eigenvalue-problems-lapack-computation/steqr.html

But in Netlib, we can not find any such information.
https://netlib.org/lapack/explore-html/d2/d24/group__aux_o_t_h_e_rcomputational_ga2463dc5fa5baab57763971a548aad789.html#ga2463dc5fa5baab57763971a548aad789

Thanks,
Younas.

**Comment by julielangou** [comment]
I think this is a recurring question we get - anybody wants to weigh in?

**Comment by mopathan** [comment]
Do we have any update on this?

**Comment by langou** [comment]
Hi Mohammad, there are two things. First I do not think we have the bandwidth to work on it. Second it is not clear we want this feature. I'll weigh on second. Depending on which variants we use the number of FLOPS for a LAPACK subroutine is not necessarily the same. Granted: Cholesky factorization is n^3/3, Cholesky inversion is n^3, LU factorization is 2/3 n^3, etc. all this assumes standard MM (i.e. no Strassen or the like). Then blocking might add some FLOPS (e.g., when we use Householder reflections) so all in all this gets messy pretty fast. I do see the value of knowing the number of FLOPS of a linear algebra LAPACK subroutine. This is a good metric to have. At this point, this falls somewhat on the back burner and is not of the highest priority. If someone has energy and time and will to work on this, we should discuss on how this would look like. Julien.

**Comment by angsch** [comment]
If someone wants to pick it up, [LAWN 41](https://netlib.org/lapack/lawnspdf/lawn41.pdf) could be a starting point. Appendix C lists the flop counts for BLAS-2 and BLAS-3 routines as well as a bunch of LAPACK routines.
